cloud/aws/deploy/website.go

```diff
@@ -0,0 +1,396 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"math"
+	"mime"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/nitrictech/nitric/cloud/aws/deploy/embeds"
+	"github.com/nitrictech/nitric/cloud/common/deploy/resources"
+	common "github.com/nitrictech/nitric/cloud/common/deploy/tags"
+	deploymentspb "github.com/nitrictech/nitric/core/pkg/proto/deployments/v1"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/cloudfront"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/s3"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	awscloudfront "github.com/aws/aws-sdk-go-v2/service/cloudfront"
+	awscloudfronttypes "github.com/aws/aws-sdk-go-v2/service/cloudfront/types"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+// createBucket - creates a new S3 bucket in AWS and tags it.
+func (a *NitricAwsPulumiProvider) createWebsiteBucket(ctx *pulumi.Context) error {
+	var err error
+
+	name := "website-bucket"
+	tags := common.Tags(a.StackId, name, resources.Bucket)
+
+	a.publicWebsiteBucket, err = s3.NewBucket(ctx, name, &s3.BucketArgs{
+		Tags: pulumi.ToStringMap(tags),
+	})
+	if err != nil {
+		return err
+	}
+
+	a.Buckets[name] = a.publicWebsiteBucket
```

Comment:
This name could conflict with another bucket with the same name in the project declared the usual way.

Reply:
Same with the tags, I wonder if this needs a different resource type or to not be stored this way at all.

Reply:
Updated